suite-native/transactions/src/components/TransactionName.tsx

```diff
@@ -0,0 +1,64 @@
+import { WalletAccountTransaction } from '@suite-common/wallet-types';
+import { Translation, TxKeyPath } from '@suite-native/intl';
+
+type TransactionNameProps = {
+    transaction: WalletAccountTransaction;
+    isPending: boolean;
+};
+
+interface GetSelfTransactionMessageByTypeProps {
+    type?: Required<WalletAccountTransaction>['cardanoSpecific']['subtype'];
+}
+
+const getSelfTransactionMessageByType = ({
+    type,
+}: GetSelfTransactionMessageByTypeProps): TxKeyPath => {
+    switch (type) {
+        case 'withdrawal':
+            return 'transactions.name.withdrawal';
+        case 'stake_delegation':
+            return 'transactions.name.stake_delegation';
+        case 'stake_registration':
+            return 'transactions.name.stake_registration';
+        case 'stake_deregistration':
+            return 'transactions.name.stake_deregistration';
+        default:
+            return 'transactions.name.self';
+    }
+};
+
+export const getTransactionName = (
+    transaction: WalletAccountTransaction,
+    isPending: boolean,
+): TxKeyPath => {
+    switch (transaction.type) {
+        case 'sent':
+            return isPending ? 'transactions.name.sending' : 'transactions.name.sent';
+        case 'recv':
+            return isPending ? 'transactions.name.receiving' : 'transactions.name.received';
+        case 'failed':
+            return 'transactions.name.failed';
+        case 'joint':
+            return 'transactions.name.joint';
+        case 'contract':
+            return 'transactions.name.contract';
+        case 'self':
+            return getSelfTransactionMessageByType({
+                type: transaction.cardanoSpecific?.subtype,
+            });
+        case 'unknown':
```

Comment:
Why list `'unknown'` when it's immediately followed by `default`? 🤔

Reply:
32c38f50ee